src/HttpAdapterFactory.php

```diff
@@ -81,4 +95,32 @@ public static function register($name, $class)
 
         self::$classes[$name] = $class;
     }
+
+    /**
+     * guesses the best matching adapter
+     *
+     * @param  array                $preferred
+     * @return HttpAdapterInterface
+     * @throws HttpAdapterException
+     */
+    public static function guess($preferred = array())
+    {
+        $available = function ($class) {
```

Comment:
What about introducing an `has` method which returns a boolean in order to detect if an adapter is available?

Something like:

``` php
HttpAdapterFactory::has('buzz');
```

Then, it would simplify your algo and added a feature for end users.